src/OpenTelemetry/Resources/ResourceBuilder.cs

```diff
@@ -0,0 +1,93 @@
+// <copyright file="ResourceBuilder.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+
+namespace OpenTelemetry.Resources
+{
+    /// <summary>
+    /// Contains methods for building <see cref="Resource"/> instances.
+    /// </summary>
+    public class ResourceBuilder
+    {
+        private readonly List<Resource> resources = new List<Resource>();
+
+        /// <summary>
+        /// Add a <see cref="Resource"/> to the builder.
+        /// </summary>
+        /// <param name="resource"><see cref="Resource"/>.</param>
+        /// <returns><see cref="ResourceBuilder"/> for chaining.</returns>
+        public ResourceBuilder AddResource(Resource resource)
+        {
+            if (resource == null)
+            {
+                throw new ArgumentNullException(nameof(resource));
+            }
+
+            this.resources.Add(resource);
+
+            return this;
+        }
+
+        /// <summary>
+        /// Clears the <see cref="Resource"/>s added to the builder.
+        /// </summary>
+        /// <returns><see cref="ResourceBuilder"/> for chaining.</returns>
+        public ResourceBuilder Clear()
+        {
+            this.resources.Clear();
+
+            return this;
+        }
+
+        /// <summary>
+        /// Build a merged <see cref="Resource"/> from all the <see cref="Resource"/>s added to the builder.
+        /// </summary>
+        /// <returns><see cref="Resource"/>.</returns>
+        public Resource Build()
+        {
+            Resource finalResource = Resource.Empty;
+
+            foreach (Resource resource in this.resources)
+            {
+                finalResource = finalResource.Merge(resource);
+            }
+
+            return finalResource;
+        }
+
+        internal static ResourceBuilder CreateDefault()
+            => new ResourceBuilder().AddTelemetrySdk().AddEnvironmentVariableDetector();
```

Comment:
Its not yet very clear from spec that this must be the default. Specifically, we don't know if the ENV variable detector should be there by default. Lets defer from making that part of default?

Reply:
I was just looking at [the spec](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/resource/sdk.md#detecting-resource-information-from-the-environment) it says:

> The SDK MUST extract information from the OTEL_RESOURCE_ATTRIBUTES environment variable and merge this, as the secondary resource, with any resource information provided by the user, i.e. the user provided resource information has higher priority.

Reply:
https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/sdk-environment-variables.md The overall Env variable guidance says this is NOT must.
Its best to be defensive now. We can add ENV variable to default in the future, if spec confirms so.

Reply:
Removed the EnvVars from the default builder.